x-pack/plugins/cases/server/client/utils.ts

```diff
@@ -527,3 +531,27 @@ export const convertSortField = (sortField: string | undefined): SortFieldCase =
       return SortFieldCase.createdAt;
   }
 };
+
+export const constructSearchById = (
+  search: string,
+  spaceId: string,
+  savedObjectsSerializer: ISavedObjectsSerializer
+) => {
+  // import { version as uuidVersion, validate as uuidValidate } from 'uuid';
+  const uuidValidate = (foobar: string) => foobar.length === 36; // uuid length
+  const uuidVersion = (foobar: string) => 4;
+
+  if (search !== '' && uuidValidate(search) && uuidVersion(search) === 4) {
+    const rawId = savedObjectsSerializer.generateRawId(
+      spaceIdToNamespace(spaceId),
+      CASE_SAVED_OBJECT,
+      search
+    );
+
+    return {
+      search: `${search} ${rawId}`,
+      rootSearchFields: ['_id'],
+    };
+  }
+  return {};
```

Comment:
I think that `constructSearchById` should be responsible for the `search` in all use cases. It will lead to confusion if it only handles the UUID use case and not the rest. I suggest returning `{ search }` if the search term is not a UUID and is defined. if `search` is `undefined` it can return `undefined`.